packages/atomic/src/components/atomic-no-results/atomic-no-results.tsx

```diff
@@ -102,7 +102,7 @@ export class AtomicNoResults {
         style="primary"
         part="cancel-button"
         text={this.bindings.i18n.t('cancel-last-action')}
-        onClick={() => this.history.back()}
+        onClick={() => this.history.backOnNoResults()}
```

Comment:
Question I had when implementing this, is the history manager controller really necessary? Or we can use the browser history to achieve this?

Reply:
> Question I had when implementing this, is the history manager controller really necessary? Or we can use the browser history to achieve this?

The difference with the history manager is that it only creates "snapshot" for "successful" state (those that returned results and that did not return a query error).

Versus browser history, which could very well just put you in a "loop" with no results or with query errors.